msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerContentProviderStrategy.java

```diff
@@ -0,0 +1,487 @@
+        } else {
+            final String message = "Failed to get result from Broker Content Provider, cursor is null";
+            Logger.error(TAG + methodName, message, null);
+            Telemetry.emit(
+                    new BrokerEndEvent()
+                            .putAction(methodName)
+                            .isSuccessful(false)
+                            .putErrorDescription(message)
+            );
+            throw new BrokerCommunicationException(message, null);
+        }
+    }
+
+    @Override
+    String hello(@NonNull CommandParameters parameters) throws BaseException {
+        final String methodName = "helloWithContentProvider";
+
+        return performContentProviderOperation(parameters, new ContentProviderOperation<CommandParameters, String>() {
+            @Override
+            public String getRequestString(CommandParameters parameters) {
+                return mRequestAdapter.getRequestStringForHello(parameters);
+            }
+
+            @Override
+            public String[] getOptions() {
+                return null; // no additional options to be sent for hello
+            }
+
+            @Override
+            public String getMethodName() {
+                return methodName;
+            }
+
+            @Override
+            public String getUriPath() {
+                return HELLO_PATH;
+            }
+
+            @Override
+            public String getResultFromBundle(Bundle resultBundle) throws BaseException {
+                return mResultAdapter.verifyHelloFromResultBundle(resultBundle);
+            }
+        });
+    }
+
+    @Override
+    Intent getBrokerAuthorizationIntent(@NonNull final InteractiveTokenCommandParameters parameters,
+                                        @Nullable final String negotiatedBrokerProtocolVersion)
+            throws BaseException {
+        final String methodName = "getBrokerAuthorizationIntentForContentProvider";
+
+        return performContentProviderOperation(parameters, new ContentProviderOperation<InteractiveTokenCommandParameters, Intent>() {
+
+            @Override
+            public String getRequestString(InteractiveTokenCommandParameters parameters) {
+                return null; // broker returns us an intent based on calling uid , no request string needed
+            }
+
+            @Override
+            public String[] getOptions() {
+                return new String[]{negotiatedBrokerProtocolVersion};
+            }
+
+            @Override
+            public String getMethodName() {
+                return methodName;
+            }
+
+            @Override
+            public String getUriPath() {
+                return ACQUIRE_TOKEN_INTERACTIVE_PATH;
+            }
+
+            @Override
+            public Intent getResultFromBundle(Bundle resultBundle) throws BaseException {
+                final Bundle requestBundle = mRequestAdapter.getRequestBundleForAcquireTokenInteractive(
```

Comment:
call BrokerBaseStrategy.completeInteractiveRequestIntent()?

Reply:
(and we might want to move this function to MsalBrokerResultAdapter too)

Reply:
moved it to `MsalBrokerResultAdapter`, we don't need to call `completeInteractiveRequestIntent` , as we construct the intent here using the bundle